packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -665,8 +665,8 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
       if (
         await observer.didPushRouteInformation(
           RouteInformation(
-            location: routeArguments['location'] as String,
-            state: routeArguments['state'] as Object,
+            location: routeArguments['location'] as String?,
```

Comment:
Hi, sorry i didn't notice this with my last review, I think the `location` should always be none null, otherwise didPushRouteInformation does not make sense.

Reply:
Ok, removed